galley/pkg/runtime/projections/serviceentry/convert/convert.go

```diff
@@ -43,7 +42,7 @@ func Annotations(serviceAnnotations resource.Annotations) resource.Annotations {
 }
 
 // Service converts a k8s Service to a networking.ServiceEntry
-func Service(spec *coreV1.ServiceSpec, metadata resource.Metadata, fullName resource.FullName, domainSuffix string) *networking.ServiceEntry {
+func Service(spec *coreV1.ServiceSpec, metadata resource.Metadata, fullName resource.FullName, domainSuffix string, out *networking.ServiceEntry) {
```

Comment:
It is not immediately obvious why out is needed. It will look like a good candidate to refactor out to someone who is not familiar with the code. Adding a comment about why out is needed will help avoid spurious refactoring requests and removal of the optimization.